packages/tracing/src/browser/web-vitals/getLCP.ts

```diff
@@ -46,30 +46,37 @@ export const getLCP = (onReport: ReportHandler, reportAllChanges = false): void
 
     // If the page was hidden prior to paint time of the entry,
     // ignore it and mark the metric as final, otherwise add the entry.
-    if (value < firstHidden.timeStamp) {
+    if (value < visibilityWatcher.firstHiddenTime) {
       metric.value = value;
       metric.entries.push(entry);
-    } else {
-      metric.isFinal = true;
     }
 
-    report();
+    if (report) {
+      report();
+    }
   };
 
   const po = observe('largest-contentful-paint', entryHandler);
 
   if (po) {
-    report = bindReporter(onReport, metric, po, reportAllChanges);
+    report = bindReporter(onReport, metric, reportAllChanges);
 
-    const onFinal = (): void => {
-      if (!metric.isFinal) {
+    const stopListening = (): void => {
+      if (!reportedMetricIDs.has(metric.id)) {
```

Comment:
Is it possible that this will never disconnect due to this specific `metric.id` being missing?

Reply:
Shouldn't be the case.